librecad/src/actions/lc_abstractactionwithpreview.cpp

```diff
@@ -149,6 +149,23 @@ bool LC_AbstractActionWithPreview::isAcceptSelectedEntityToTriggerOnInit([[maybe
 void LC_AbstractActionWithPreview::doPerformOriginalEntitiesDeletionOnInitTrigger([[maybe_unused]]QList<RS_Entity *> &list){}
 
 
+void LC_AbstractActionWithPreview::updateSnapperAndCoordinateWidget(QMouseEvent* e, [[maybe_unused]]int status){
+    // todo - actually, this is a bit ugly to call snap point  - yet as side effect, it will draw snapper and update coordinates widget..
+    RS_Vector mouse = snapPoint(e);
+}
+
+/**
+ * Explicitly updates coordinate widget by current mouse position.
+ * Method is useful for actions states that do not call snapPoint() method on mouse move (which, in turn, updates the widget)
+ * @param e
+ */
+void LC_AbstractActionWithPreview::doUpdateCoordinateWidgetByMouse(QMouseEvent* e){
+    RS_Vector mouse = graphicView->toGraph(e->x(), e->y());
```

Comment:
Please use e->position ()

X, y methods are deprecated

Reply:
One more thing, we should try to minimize the header dependency of base classes.

Decoupling headers is good in long run.

To remove the "rs_vector" dependency, we can keep pointer members in the base class and use accessors, like:
```
class RS_Vector;
class Base {

public:
const RS_Vector& getPoint() const;
RS_Vector& getPoint();
void setPoint(const RS_Vector&);

private:
std::unique_ptr<RS_Vector> m_aPoint;
};

// in cpp

const RS_Vector& Base::getPoint() const 
{
     return m_aPoint != nullptr ? *m_aPoint: RS_Vector{};
}

RS_Vector& Base::getPoint()
{
    assert(m_aPoint != nullptr);
    if (m_aPoint!=nullptr)
        return *m_aPoint;
    static RS_Vector shouldnotUse{};
    return shouldnotUse;
}

void Base::setPoint(const RS_Vector& newValue)
{
    if (m_aPoint == nullptr)
        m_aPoint = std::make_unique<RS_Vector>();
    *m_aPoint = newValue;
}
```